huntsman/observatory.py

```diff
@@ -338,67 +338,57 @@ def _create_autoguider(self):
 
         self.autoguider = guider
 
-    def _create_flat_field_observation(self, alt=None, az=None,
-                                       dither_pattern_offset=5 * u.arcmin,
-                                       dither_random_offset=0.5 * u.arcmin,
-                                       n_positions=9, filter_name=None,
-                                       ):
+    def _create_flat_field_observation(self, alt=None, az=None, *args, **kwargs):
+
+        # Specify coordinates for flat field
         if alt is None and az is None:
+            self.logger.debug('Using flat-field alt/az from config.')
             flat_config = self.config['flat_field']['twilight']
             alt = flat_config['alt']
             az = flat_config['az']
+        flat_coords = utils.altaz_to_radec(alt=alt, az=az, location=self.earth_location,
+                                           obstime=utils.current_time())
+        self.logger.debug(f'Flat field coordinates: {flat_coords}')
+
+        # Create the Observation object
+        obs = DitheredFlatObservation(position=flat_coords, *args, **kwargs)
+        obs.seq_time = utils.current_time(flatten=True)
+        self.logger.debug("Flat-field observation: {}".format(obs))
+
+        return obs
+
+    def _calculate_flat_exptime(self, previous_exptime, elapsed_time, target_adu,
+                                mean_counts, min_exptime):
+        """Calculate the next exposure time for the flat fields, accounting
+        for changes in sky brightness."""
+        exptime = previous_exptime * (target_adu / mean_counts)
+
+        sky_factor = 2.0 ** (elapsed_time / 180.0)
+        if self.past_midnight:
```

Comment:
Yep it would. But it seems a bit redundant since `past_midnight` is already a property. It should still be possible to test by modifying the time as in other tests.

Reply:
It will make it *way* easier to test and modify in the future if it is decoupled from the observatory. Easy enough to do now but could leave for future PR.